bin/host/src/backend/online.rs

```diff
@@ -97,8 +97,9 @@ where
     async fn route_hint(&self, hint: String) -> PreimageOracleResult<()> {
         trace!(target: "host_backend", "Received hint: {hint}");
 
-        let parsed_hint =
-            hint.parse::<Hint<C::HintType>>().map_err(|_| PreimageOracleError::KeyNotFound)?;
+        let parsed_hint = hint
+            .parse::<Hint<C::HintType>>()
+            .map_err(|e| PreimageOracleError::Other(format!("failed to parse hint: {e}")))?;
```

Comment:
can we do this properly and add another variant instead pls

Reply:
+ a unit test